File: patient/src/main/java/com/openspace/PatientManagement/service/CurrencyService.java
package com.openspace.PatientManagement.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.openspace.Model.DoctorManagement.Currency;

public interface CurrencyService {

	void addCurrency(Currency currency);

	List<Currency> getAllCurrencys();

	void updateCurrency(Currency currency);

	void deleteCurrency(Long id);

	List<Currency> getAllCurrencysByAdmin(String username);

	Page<Currency> getAllCurrencysByAdminPage(String username, int page, int size);
}

File: portal/src/main/resources/static/service/finance/currencyService.js
function currencyService(PATIENT_MODULE_CONFIG,MODULE_CONFIG, $http) {
	this.addCurrency = function(currency) {
	return 	$http.post(PATIENT_MODULE_CONFIG.ADD_CURRENCY(),currency);
	};
	this.getAllCurrencys=function(email){
		return $http.get(PATIENT_MODULE_CONFIG.GET_ALL_CURRENCYS(email));
	};
	this.getAllCurrencyList=function(email,page, size){
		return $http.get(PATIENT_MODULE_CONFIG.GET_ALL_CURRENCY_LIST(email,page, size));
	};
	this.updateCurrency=function(currency){
		return $http.put(PATIENT_MODULE_CONFIG.UPDATE_CURRENCY(),currency);
	};
	this.deleteCurrency=function(id){
		return $http.delete(PATIENT_MODULE_CONFIG.DELETE_CURRENCY(id));
	};
	
}
angular.module('HealthApplication').service('currencyService', currencyService);